test/integration/tas/tas_test.go

```diff
@@ -839,3 +840,56 @@ var _ = ginkgo.Describe("Topology Aware Scheduling", ginkgo.Ordered, func() {
 		})
 	})
 })
+
+var _ = ginkgo.Describe("Topology validations", func() {
+	ginkgo.When("Creating a Topology", func() {
+		ginkgo.DescribeTable("Validate Topology on creation", func(topology *kueuealpha.Topology, matcher types.GomegaMatcher) {
+			err := k8sClient.Create(ctx, topology)
+			if err == nil {
+				defer func() {
+					util.ExpectObjectToBeDeleted(ctx, k8sClient, topology, true)
+				}()
+			}
+			gomega.Expect(err).Should(matcher)
+		},
+			ginkgo.Entry("valid Topology",
+				testing.MakeTopology("valid").Levels([]string{corev1.LabelHostname}).Obj(),
+				gomega.Succeed()),
+			ginkgo.Entry("invalid levels",
+				testing.MakeTopology("invalid-level").Levels([]string{"@invalid"}).Obj(),
+				testing.BeInvalidError()),
+		)
+	})
+
+	ginkgo.When("Updating a Topology", func() {
+		ginkgo.DescribeTable("Validate Topology on update", func(topology *kueuealpha.Topology, updateTopology func(topology *kueuealpha.Topology), matcher types.GomegaMatcher) {
+			err := k8sClient.Create(ctx, topology)
+			if err == nil {
+				defer func() {
+					util.ExpectObjectToBeDeleted(ctx, k8sClient, topology, true)
+				}()
```

Comment:
I think it will be more readable if we move the defer block below `gomega.Expect(err).Should(gomega.Succeed())`